QModManager/API/SMLHelper/Crafting/TechData.cs

```diff
@@ -0,0 +1,146 @@
+﻿namespace QModManager.API.SMLHelper.Crafting
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// A class that fully describes a recipe for a <see cref="TechType"/> identified item.
+    /// </summary>
+    /// <seealso cref="ITechData" />
+    public class TechData : ITechData
+    {
+        /// <summary>
+        /// Gets or sets the how many copies of the item are created when crafting this recipe.
+        /// </summary>
+        /// <value>
+        /// The quantity of the item this recipe yields.
+        /// </value>
+        public int craftAmount { get; set; }
+
+        /// <summary>
+        /// Gets the number of different ingredients for this recipe.
+        /// </summary>
+        /// <value>
+        /// The number of ingredients for this recipe.
+        /// </value>
+        public int ingredientCount => Ingredients.Count;
+
+        /// <summary>
+        /// Gets the number of items linked to this recipe.
+        /// </summary>
+        /// <value>
+        /// The number of linked items.
+        /// </value>
+        public int linkedItemCount => LinkedItems.Count;
+
+        /// <summary>
+        /// The list of ingredients required for this recipe.
+        /// </summary>
+        public List<Ingredient> Ingredients = new List<Ingredient>();
+
+        /// <summary>
+        /// The items that will also be created when this recipe is crafted.
+        /// </summary>
+        public List<TechType> LinkedItems = new List<TechType>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TechData"/> class a custom recipe.
+        /// </summary>
+        public TechData() { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TechData"/> class for a custom recipe with a list of ingridients.
+        /// </summary>
+        /// <param name="ingredients">The ingredients.</param>
+        public TechData(List<Ingredient> ingredients)
+        {
+            Ingredients = ingredients;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TechData"/> class for a custom recipe with a collection of ingridients.
+        /// </summary>
+        /// <param name="ingredients">The ingredients.</param>
+        public TechData(params Ingredient[] ingredients)
```

Comment:
new constructor idea: `public TechData(params TechType[] types)`

Reply:
Assuming you mean `public TechData(params TechType[] ingredients)`, sure.  
That would be fine for many simple recipe scenarios.